clash-prelude/src/Clash/Signal/Delayed.hs

```diff
@@ -5,20 +5,31 @@ License    :  BSD2 (see the file LICENSE)
 Maintainer :  Christiaan Baaij <christiaan.baaij@gmail.com>
 -}
 
-{-# LANGUAGE CPP                 #-}
-{-# LANGUAGE DataKinds           #-}
-{-# LANGUAGE FlexibleContexts    #-}
-{-# LANGUAGE KindSignatures      #-}
-{-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE TypeOperators       #-}
+{-# LANGUAGE CPP                  #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE UndecidableInstances #-}
+#if __GLASGOW_HASKELL__ >= 806
```

Comment:
This is for GHC 8.0.6, right? We don't support <8.2, so I think you can remove this.

Reply:
Actually kind of confusingly, 806 refers to [8.6.0 (see here)](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/intro.html#ghc-version-numbering-policy). This case is needed for 8.6.0 and above as GHC starts interpreting all unqualified cases of * as `Type` (interfering with type level multiplication). We could also enable NoStarIsType in the cabal config or use ugly qualifications, but it looks like Clash is using this same construct elsewhere, and a general change like that would be better in its own PR? ([8.6.0 StarIsType migration notes](https://ghc.haskell.org/trac/ghc/wiki/Migration/8.6#StarIsType))

Reply:
I see, thanks for the pointer! And I agree, that change is better suited for another PR.